### ISSUE TITLE
How to get player.stop() to work

### ISSUE THREAD

**Issue by andevellicus** [node-xbmc/node-xbmc#12]
I can get most of the code to work, however for some reason whenever I try to invoke the player.stop() or player.playPause(), I get an error. The code I'm using is from the examples:

```javascript
(function() {
  var TCPConnection, XbmcApi, config, connection, file, xbmcApi, _ref;

  _ref = require('xbmc'), TCPConnection = _ref.TCPConnection, XbmcApi = _ref.XbmcApi;

 

  connection = new TCPConnection({
    host: my_kodi_ip,
    port: 9090,
    verbose: true
  });

  xbmcApi = new XbmcApi({
    silent: true,
    connection: connection
  });

  xbmcApi.player.stop();

}).call(this);
```

The error I get is:

```
~/nodejs/node_modules/xbmc/node_modules/node-promise/promise.js:204
          throw error;
                ^
TypeError: Cannot call method 'send' of undefined
    at ~/nodejs/node_modules/xbmc/lib/Player.js:90:24
    at ~/nodejs/node_modules/xbmc/lib/Player.js:55:18
    at Object._onImmediate (~/nodejs/node_modules/xbmc/node_modules/node-promise/promise.js:164:27)
    at processImmediate [as _immediateCallback] (timers.js:330:15)
```

Not quite sure what I'm doing wrong... would appreciate any help.

**Comment by moul** [comment]
Fixed in upstream and released as version v0.6.0 on NPM

Can you give a new try ?

**Comment by frisco82** [comment]
This is still causing an exception because of using that `||` to check for one result or another.
if `data.result[0].playerid == 0` then it will evaluate as `false` and go for `data.player` which doesn't exist.

**Comment by moul** [comment]
ok good catch, do you have an example of code triggering this ?

**Comment by frisco82** [comment]
The code is as simple as calling ```api.player.stop()```

Error when there is no active player (I am doing an open before the closes ends but it doesn't matter for the error).
```
  xbmc:Player stop +0ms
  xbmc:Player getActivePlayers +1ms
  xbmc:XbmcApi send +0ms Player.GetActivePlayers {}
  xbmc:TCPConnection send +0ms {"method":"Player.GetActivePlayers","params":{}}
  xbmc:TCPConnection isActive +2ms
  xbmc:TCPConnection publish +1ms send {"method":"Player.GetActivePlayers","params":{},"id":"__id4","jsonrpc":"2.0"}
  xbmc:Player openFile +0ms /media/Seagate Expansion Drive/Prueba/inicio.mp4 {}
  xbmc:Player open +1ms { file: '/media/Seagate Expansion Drive/Prueba/inicio.mp4' } {}
  xbmc:XbmcApi send +0ms Player.Open {"item":{"file":"/media/Seagate Expansion Drive/Prueba/inicio.mp4"},"options":{}}
  xbmc:TCPConnection send +0ms {"method":"Player.Open","params":{"item":{"file":"/media/Seagate Expansion Drive/Prueba/inicio.
mp4"},"options":{}}}
  xbmc:TCPConnection isActive +0ms
  xbmc:TCPConnection publish +2ms send {"method":"Player.Open","params":{"item":{"file":"/media/Seagate Expansion Drive/Prueba
/inicio.mp4"},"options":{}},"id":"__id5","jsonrpc":"2.0"}
  express:view lookup "index.jade" +266ms
  express:view stat "D:\Node\prueba\views\index.jade" +1ms
  express:view render "D:\Node\prueba\views\index.jade" +0ms
  morgan log request +110ms
POST / 200 411.611 ms - 1342
  xbmc:TCPConnection onData +2ms
  xbmc:TCPConnection publish +0ms data {"id":"__id4","jsonrpc":"2.0","result":[]}
  xbmc:TCPConnection publish +0ms data {"jsonrpc":"2.0","method":"GUI.OnScreensaverDeactivated","params":{"data":{"shuttingdow
n":false},"sender":"xbmc"}}
  xbmc:TCPConnection publish +1ms notification {"jsonrpc":"2.0","method":"GUI.OnScreensaverDeactivated","params":{"data":{"shu
ttingdown":false},"sender":"xbmc"}}
  xbmc:Notifications delegate +0ms { jsonrpc: '2.0',
  method: 'GUI.OnScreensaverDeactivated',
  params: { data: { shuttingdown: false }, sender: 'xbmc' } }
  xbmc:Notifications screensaverdeactivated +1ms
  xbmc:TCPConnection publish +1ms data {"id":"__id5","jsonrpc":"2.0","result":"OK"}
D:\Node\prueba\node_modules\xbmc\node_modules\node-promise\promise.js:204
          throw error;
          ^

TypeError: Cannot read property 'playerid' of undefined
    at D:\Node\prueba\node_modules\xbmc\lib\Player.js:87:57
    at D:\Node\prueba\node_modules\xbmc\lib\Player.js:55:18
    at Immediate._onImmediate (D:\Node\prueba\node_modules\xbmc\node_modules\node-promise\promise.js:164:27)
    at processImmediate [as _immediateCallback] (timers.js:368:17)
````

Error stopping audio file, sometimes it works sometimes it just does not send the ```palyerid``` param.
```
  xbmc:Player stop +1ms
  xbmc:Player getActivePlayers +1ms
  xbmc:XbmcApi send +0ms Player.GetActivePlayers {}
  xbmc:TCPConnection send +0ms {"method":"Player.GetActivePlayers","params":{}}
  xbmc:TCPConnection isActive +1ms
  xbmc:TCPConnection publish +0ms send {"method":"Player.GetActivePlayers","params":{},"id":"__id18","jsonrpc":"2.0"}
  morgan log request +1ms
GET /control/stop 304 19.447 ms - -
  xbmc:TCPConnection onData +3ms
  xbmc:TCPConnection publish +1ms data {"id":"__id18","jsonrpc":"2.0","result":[{"playerid":0,"type":"audio"}]}
  xbmc:XbmcApi send +1ms Player.Stop {}
  xbmc:TCPConnection send +0ms {"method":"Player.Stop","params":{}}
  xbmc:TCPConnection isActive +0ms
  xbmc:TCPConnection publish +0ms send {"method":"Player.Stop","params":{},"id":"__id19","jsonrpc":"2.0"}
  xbmc:TCPConnection onData +4ms
  xbmc:TCPConnection onError +2ms {"data":{"error":{"code":-32602,"data":{"method":"Player.Stop","stack":{"message":"Missing p
arameter","name":"playerid","type":"integer"}},"message":"Invalid params."},"id":"__id19","jsonrpc":"2.0"}}
  xbmc:TCPConnection publish +1ms error {"data":{"error":{"code":-32602,"data":{"method":"Player.Stop","stack":{"message":"Mis
sing parameter","name":"playerid","type":"integer"}},"message":"Invalid params."},"id":"__id19","jsonrpc":"2.0"}}

D:\Node\prueba\node_modules\xbmc\node_modules\node-promise\promise.js:204
          throw error;
          ^
```

**Comment by moul** [comment]
I just fixed the issue, can you try again with the latest released version (0.6.1 on NPM) ?
